src/main/kotlin/io/github/cottonmc/jsonfactory/gens/ContentGeneratorTemplate.kt

```diff
@@ -0,0 +1,18 @@
+package io.github.cottonmc.jsonfactory.gens
+
+import io.github.cottonmc.jsonfactory.data.Identifier
+import io.github.cottonmc.jsonfactory.output.Output
+
+/**
+ * Template for content generators
+ * @see ContentGenerator
+ * */
+interface ContentGeneratorTemplate {
```

Comment:
I think this is not needed; `DecoratedContentGenerator` should instead pass the properties of the delegate as constructor arguments.

Reply:
Alright. A possible solution would be:
- `ContentGenerator` ⇒ `AbstractContentGenerator`
- `ContentGeneratorTemplate` ⇒ `ContentGenerator`

Reply:
I did not wanted to touch that class yet, but doing it now.

Reply:
And I'll update the builder, because it is not using it yet.